hackathon_site/hardware/test_api.py

```diff
@@ -864,6 +864,89 @@ def test_successful_post(self):
             self.assertEqual(final_response[attribute], self.request_data[attribute])
 
 
+class IncidentDetailViewGetTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.profile = Profile.objects.create(user=self.user)
+        self.team = Team.objects.create()
+        self.order = Order.objects.create(
+            status="Cart",
+            team=self.team,
+            request={"hardware": [{"id": 1, "quantity": 2}, {"id": 2, "quantity": 3}]},
+        )
+        self.hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.other_hardware = Hardware.objects.create(
+            name="other",
+            model_number="otherModel",
+            manufacturer="otherManufacturer",
+            datasheet="/datasheet/location/other",
+            quantity_available=10,
+            max_per_team=10,
+            picture="/picture/location/other",
+        )
+        self.order_item = OrderItem.objects.create(
+            order=self.order, hardware=self.hardware,
+        )
+
+        self.order_item2 = OrderItem.objects.create(
+            order=self.order, hardware=self.other_hardware,
+        )
+
+        self.incident = Incident.objects.create(
+            state="Broken",
+            description="Description",
+            order_item=self.order_item,
+            time_occurred=datetime(2022, 8, 8, tzinfo=settings.TZ_INFO),
+        )
+
+        self.another_incident = Incident.objects.create(
+            state="Missing",
+            description="Description",
+            order_item=self.order_item2,
+            time_occurred=datetime(2022, 1, 1, tzinfo=settings.TZ_INFO),
+        )
+        self.view_permissions = Permission.objects.filter(
+            content_type__app_label="hardware", codename="view_incident"
+        )
+
+        self.view = reverse("api:hardware:incident-detail", args=[self.incident.id])
+
+    def test_user_not_logged_in(self):
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_has_no_profile(self):
```

Comment:
here instead, you should test that a user should not be able to use this api if they have a profile but no view permissions.
you can use this line in your setup to create a profile in this function only: 
`Profile.objects.create(user=self.user)`